### DISCUSSION TITLE
API integration or help creating a discord bot that automatize

### DISCUSSION THREAD

**Discussion by llors418** [Dadangdut33/Speech-Translate#11] [Ideas]
im just starting to learn python, i want to create a bot that lets me transcribe mettings with my friends on discord for dnd session, could i get any help, i have no idea of how big of an scope this really is

**Comment by Dadangdut33** [comment]
That is possible but might be hard to implement. 

You would need to find a way for your bot to be able to receive audio of the users in the voice channel. It would be possible for whisper to transcript more accurately if you can get audio of each user separately and then transcripting them. But if it is not possible you might need to [see this discussion](https://github.com/openai/whisper/discussions/264) because Whisper itself is actually not build for multispeaker environment.

It might cost lots of resource tho for the bot and the model to work efficiently. And you might want to separate between the bot and the transcription backend by creating a simple webserver for the transcription. With this method you can use language other than python for the bot.

My tips would be first find a reliable discord bot framework to use for your project here, then start working on the webserver.

Good luck with your ideas :D
